docs/user-journeys/users/application-developer/foundational.md

```diff
@@ -9,80 +9,151 @@ track: "USERS › APPLICATION DEVELOPER › FOUNDATIONAL"
+
+This process can be tedious and also inconvenient, especially if (2) happens in the early hours of the morning!
+
+**If you have Kubernetes set up, however, manual intervention is not as necessary.** Kubernetes will gracefully handle (3) and (4) on your behalf. As a a result, it's often referred to as a *self-healing* system.
+
+On a high-level, there are two key concepts to making this work: the *Kubernetes API* and the *Control Plane*.
+
+#### Kubernetes API
+
+For Kubernetes to be useful, it needs to know *what* sort of cluster state you want it to maintain. The {% glossary_tooltip text="Kubernetes API" term_id="kubernetes-api" %} provides a way for you to declare this desired state, in terms of API objects (such as {% glossary_tooltip text="Deployments" term_id="deployment" %}). Cluster state includes but is not limited to the following information:
 
 * The applications or other workloads to run
 * The container images for your applications and workloads
 * Allocation of network and disk resources
 
-Once you’ve set your desired state -- that is, applied your manifest or spec to your cluster -- the Kubernetes Control Plane works to make the cluster’s current state match the desired state. To do so, Kubernetes performs a variety of tasks automatically: starting or restarting containers, scaling the number of replicas of a given application, and more. The Kubernetes Control Plane consists of a collection of processes running on your cluster:
+Each API object can be described in a YAML or JSON file called a *manifest*, which can be submitted to the appropriate API server endpoint.
+
+`kubectl`, the command-line tool that allows you to easily read or modify your Kubernetes cluster, is basically just a user-friendly wrapper for the Kubernetes API. You're able to scale app instances and get node info because, under the hood, it's making API requests for you.
+
+You can read more about the Kubernetes API [here](/docs/concepts/overview/working-with-objects/kubernetes-objects/){:target="_blank"}.
+
+#### Control Plane
+
+Once you’ve set your desired state -- that is, applied your YAML manifest to your cluster -- the **Kubernetes Control Plane** works to make the cluster’s current state match the desired state.
+
+It's referred to as a "plane" because it is not a single, coherent component, but rather different processes spread across your cluster's [master node(s)](/docs/concepts/overview/components/#master-components){:target="_blank"} and [worker nodes](/docs/concepts/overview/components/#node-components){:target="_blank"}.
+
+The aforementioned Kubernetes API server actually runs as part of the Control Plane (`kube-apiserver`).
+
+The rest of the Control Plane (`kube-controller-manager`, `kube-scheduler`, `kubelet`, etc) essentially implements a "control loop". For simplicity, you can think of it as continuously asking the following:
 
```

Comment:
We need to discuss our definition of control plane. In [Kubernetes the Hard Way](https://github.com/kelseyhightower/kubernetes-the-hard-way/blob/master/docs/08-bootstrapping-kubernetes-controllers.md), Kelsey includes only three components in what he calls the control plane: kube-apiserver, kube-controller-manager, and kube-scheduler.

Also, [this topic](https://kubernetes.io/docs/concepts/overview/components/) says, "Master components provide the cluster's control plane." And it does not list kubelet among the master components. So that makes me think the author did not consider kubelet to be part of the control plane.